src/mbgl/renderer/render_orchestrator.cpp

```diff
@@ -348,12 +348,16 @@ std::unique_ptr<RenderTree> RenderOrchestrator::createRenderTree(const UpdatePar
         }
     }
 
-    for (auto& renderItem : layerRenderItems) {
-        RenderLayer& renderLayer = renderItem.layer;
-        renderLayer.prepare({renderItem.source, *imageManager, *patternAtlas, *lineAtlas, updateParameters.transformState});
+    uint32_t i = static_cast<uint32_t>(layerRenderItems.size()) - 1;
+    for (auto it = layerRenderItems.begin(); it != layerRenderItems.end(); ++it, --i) {
```

Comment:
why not keep `for (auto& renderItem : layerRenderItems)`?

Reply:
L.358 i - is index that needs to be stored to opaquePassCutOff.

Reply:
Could we similarly update `i`  inside `for (auto& renderItem : layerRenderItems)`?

Reply:
I thought it is a good idea to use [the same loop as in ](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/renderer/renderer_impl.cpp#L167)
```
        uint32_t i = static_cast<uint32_t>(layerRenderItems.size()) - 1;
        for (auto it = layerRenderItems.begin(); it != layerRenderItems.end(); ++it, --i) {
```

Not sure what you are aiming at. Would you prefer this instead?

```
        uint32_t i = static_cast<uint32_t>(layerRenderItems.size()) - 1;
        for (auto& renderItem : layerRenderItems) {
            --i;
```

Reply:
We do not need iterators per se, imo `for (auto& renderItem : layerRenderItems)` is slightly more readable than `for (auto it = layerRenderItems.begin(); it != layerRenderItems.end(); ++it, --i)`. Could be smth like
```
auto opaquePassCutOffEstimation = layerRenderItems.size();
for (auto& renderItem : layerRenderItems) {
...
if (renderTreeParameters->opaquePassCutOff == 0u) {
    --opaquePassCutOffEstimation;
    if (renderLayer.is3D()) renderTreeParameters->opaquePassCutOff = uint32_t(opaquePassCutOffEstimation);
}

}
```
However this is actually a nit, so if you prefer current approach we can proceed with it and return to this latter.